### PR TITLE
Add printer that prints CILs internal AST representation

### PR DESCRIPTION
**Pull request by michael-schwarz**
This can be helpful for understanding how `C` code is represented by `Cil`.

### CODE REVIEW

File: src/cil.ml [hunk 1]
```diff
@@ -4616,7 +4936,7 @@ class defaultCilPrinterClass : cilPrinter = object (self)
 
 end (* class defaultCilPrinterClass *)
 
-let defaultCilPrinter = new defaultCilPrinterClass
+let defaultCilPrinter = new cilAstDumpClass
```
Comment (michael-schwarz) [inline src/cil.ml:4939]: Let's not change the default one to this, we should leave it as it as.

Comment (vogler) [reply src/cil.ml:4939]: Just wanted to ask.
What's the difference to `defaultCilPrinterClass`? Does it just pretty print as valid C?

Comment (michael-schwarz) [reply src/cil.ml:4939]: Yes, it is e.g. the one we use for `--enable justcil` in Goblint.

**Comment by keremc** [comment]
Hello. Please note that this AST printer is not yet complete. For example, at the moment, it fails for any expression that is not a `PlusA` or a `Mult`. I only wrote it to get a better understanding of how CIL models the following code, which contains a variably-modified type:
```c
int main(void) {
  int n = 4;
  typedef int vla[n];
  n++;
  vla a;
  int i = sizeof(a);
}
```

**Comment by michael-schwarz** [comment]
Do we want to merge this so people can use it if they want to understand details about the representation? Or should we just close it?
I'm fine with either - @sim642 Any opinions?

**Comment by sim642** [comment]
There's `plainCilPrinterClass`, which is used by `d_plainexp`, `d_plaintype`, etc, which seems to be similar but based on a quick look has two differences:
1. It doesn't handle statements.
2. It takes some shortcuts and makes some omissions, e.g. `BinOp(PlusA,e1,e2,t)` is printed as `PlusA(e1,e2)`.

I think we can just close it for now. There hasn't been need to have something more than the plain printers much.

**Pull request closed without merging**